pkg/controllers/member/internalserviceexport/controller.go

```diff
@@ -179,9 +179,12 @@ func (r *Reconciler) reportBackConflictCondition(ctx context.Context,
 		klog.V(4).InfoS("No conflict condition to report back", "internalServiceExport", internalSvcExportRef)
 		return false, nil
 	}
-
+	// Ideally, internalServiceExport needs to track both service and serviceExport generation.
+	// But the service generation won't be populated by the controller.
+	desiredSvcExportConflictCond := internalSvcExportConflictCond.DeepCopy()
+	desiredSvcExportConflictCond.ObservedGeneration = svcExport.Generation
 	svcExportConflictCond := meta.FindStatusCondition(svcExport.Status.Conditions, string(fleetnetv1alpha1.ServiceExportConflict))
-	if reflect.DeepEqual(internalSvcExportConflictCond, svcExportConflictCond) {
+	if condition.EqualCondition(svcExportConflictCond, desiredSvcExportConflictCond) {
```

Comment:
this won't check message, any reason to change?

Reply:
reverted